client/core/core.go

```diff
@@ -4642,14 +4642,22 @@ func (c *Core) dbTrackers(dc *dexConnection) (map[order.OrderID]*trackedTrade, e
 				// tracker.cancel is set from LinkedOrder with cancelTrade.
 				continue
 			}
+			// Make sure that a taker will not prematurely send an
+			// initialization until it is confimed with the server
+			// that the match is not revoked.
+			var checkServerRevoke bool
+			if dbMatch.Side == order.Taker && dbMatch.Status == order.MakerSwapCast {
+				checkServerRevoke = true
```

Comment:
The taker hastily broadcasting their swap when the match is revoked is the mots obvious pitfall.  Now I'm wondering about maker in newly matched.  That one is going to be extremely unlikely to exists client side because maker broadcasts right away on match, so that's probably good.  Any other pitfalls?

Reply:
> what is the path that that is causing the hasty swap

I think the ticker in listen. Run > initialize > connectAccount > connectDEX > listen > tick

Now setting the field on disconnect, so that it will check again on reconnect.